database/database.go

```diff
@@ -486,3 +489,75 @@ func (s *DatabaseService) DeleteExecutionPayloads(idFirst, idLast uint64) error
 	_, err := s.DB.Exec(query, idFirst, idLast)
 	return err
 }
+
+func (s *DatabaseService) UpsertBuilderDemotion(submitBlockRequest *types.BuilderSubmitBlockRequest, signedBlindedBeaconBlock *types.SignedBlindedBeaconBlock, signedValidatorRegistration *types.SignedValidatorRegistration) error {
```

Comment:
I led you astray here. We need to save the unblinded signed beacon as part of the demotion (not `signedBlindedBeaconBlock`).

Reply:
done! there already existed a function to convert the SignedBlinded to a SignedBeacon: `SignedBlindedBeaconBlockToBeaconBlock`, so i was able to make use of that :-)